### PR TITLE
Add a pull request template

### PR DESCRIPTION
**Pull request by aeisenberg**
This PR adds a pull request template. It is sparse right now. The main goal is to remind query authors to check for autofix changes when they add new queries or update existing ones.

### CODE REVIEW

**Comment by aeisenberg** [comment]
I'm sure we can fill this out with a lot more. For now, though, I just want a non-intrusive way to remind about autofix. Suggestions on other things to add would be appreciated.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+### Pull Request checklist
+
+- [ ] Add a change if necessary. See [the documentation](https://github.com/github/codeql/blob/main/docs/change-notes.md).
+- [ ] If this PR makes any changes to `.ql`, `.qll`, or `.qhelp` files, make sure that autofixes generated based on these changes are valid. See [the documentation](https://github.com/github/codeql-team/blob/main/docs/best-practices/validating-autofix-for-query-changes.md) (internal access required).
```
Comment (erik-krogh) [inline .github/pull_request_template.md:4]:
```suggestion
- [ ] If this PR makes significant changes to `.ql`, `.qll`, or `.qhelp` files, make sure that autofixes generated based on these changes are valid. See [the documentation](https://github.com/github/codeql-team/blob/main/docs/best-practices/validating-autofix-for-query-changes.md) (internal access required).
```

**Review by erik-krogh — COMMENTED** [review]
We should probably also add: 
- [ ] Add tests. 
- [x] Use DCA

**Review by erik-krogh — APPROVED** [review]
LGTM, but Esben should also have a look.

**Comment by aeisenberg** [comment]
Adding codeql Engineering Managers as reviewers since we discussed this PR this morning.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+### Pull Request checklist
+
+#### All query authors
+
+- [ ] Add a change note if necessary. See [the documentation](https://github.com/github/codeql/blob/main/docs/change-notes.md) in this repository.
+- [ ] All new queries have appropriate `.qhelp`. See [the documentation](https://github.com/github/codeql/blob/mainπ /docs/query-help-style-guide.md) in this repository.
```
Comment (adityasharad) [inline .github/pull_request_template.md:6]: Rogue symbol!
```suggestion
- [ ] All new queries have appropriate `.qhelp`. See [the documentation](https://github.com/github/codeql/blob/main/docs/query-help-style-guide.md) in this repository.
```

**Review by adityasharad — DISMISSED** [review]
Minor typo, otherwise happy with this as a starting point we can refine over time.

File: .github/pull_request_template.md (hunk 2 above)
Comment (aeisenberg) [reply .github/pull_request_template.md]: Wild. Not sure how that slipped in.

**Review by erik-krogh — APPROVED** [review]
Looks good to me. But Esben should still have a look.

**Comment by oscarsj** [comment]
mentioning @esbena in case this was lost in the notifications maelstrom

**Comment by aeisenberg** [comment]
I'll keep this open until Tuesday so I can give others a chance to review.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+### Pull Request checklist
+
+#### All query authors
+
+- [ ] Add a change note if necessary. See [the documentation](https://github.com/github/codeql/blob/main/docs/change-notes.md) in this repository.
```
Comment (esbena) [inline .github/pull_request_template.md:5]: Inconsistent style. I'm not going to discuss which of the two styles that is preferable.

```suggestion
- [ ] Change notes are added if necessary. See [the documentation](https://github.com/github/codeql/blob/main/docs/change-notes.md) in this repository.
```

This matches the declarative style of the other bullets, especially "QL tests are added..." one.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+- [ ] All new queries have appropriate `.qhelp`. See [the documentation](https://github.com/github/codeql/blob/main/docs/query-help-style-guide.md) in this repository.
+- [ ] QL tests are added if necessary. See [Testing custom queries](https://docs.github.com/en/code-security/codeql-cli/using-the-advanced-functionality-of-the-codeql-cli/testing-custom-queries) in the GitHub documentation.
+- [ ] New and changed queries have correct query metadata. See [the documentation](https://github.com/github/codeql/blob/main/docs/query-metadata-style-guide.md) in this repository.
+
+#### Internal query authors only
+
+- [ ] If this PR makes significant changes to `.ql`, `.qll`, or `.qhelp` files, make sure that autofixes generated based on these changes are valid. See [the documentation](https://github.com/github/codeql-team/blob/main/docs/best-practices/validating-autofix-for-query-changes.md) (internal access required).
+- [ ] Test your changes [at scale](https://github.com/github/codeql-dca/) (internal access required).
```
Comment (esbena) [inline .github/pull_request_template.md:13]: These two bullets might need a reformulation to match the style above.

**Review by esbena — APPROVED** [review]

**Pull request merged**